tests/test_debugging.py

```diff
@@ -503,6 +504,144 @@ def test_has_active_device(self):
+    )
+    def test_execute(self, dev, tape, expected_result):
+        """Test that the _execute method works as expected."""
+        PLDB.add_device(dev)
+        executed_results = PLDB._execute((tape,))
+        assert qnp.allclose(expected_result, executed_results)
+        PLDB.reset_active_dev()
+
+
+def test_tape():
+    """Test that we can access the tape from the active queue."""
+    with qml.queuing.AnnotatedQueue() as queue:
+        qml.X(0)
+        [qml.Hadamard(i) for i in range(3)]
+        qml.Y(1)
+        qml.Z(0)
+        qml.expval(qml.Z(0))
+
+        executed_tape = qml.debugging.tape()
+
+    expected_tape = qml.tape.QuantumScript.from_queue(queue)
+    assert qml.equal(expected_tape, executed_tape)
+
+
+@pytest.mark.parametrize("measurement_process", (qml.expval(qml.Z(0)), qml.state(), qml.probs()))
+@patch.object(PLDB, "_execute")
+def test_measure(mock_method, measurement_process):
+    """Test that the private measure function doesn't modify the active queue"""
+    with qml.queuing.AnnotatedQueue() as queue:
+        ops = [qml.X(0), qml.Y(1), qml.Z(0)] + [qml.Hadamard(i) for i in range(3)]
+        measurements = [qml.expval(qml.X(2)), qml.state(), qml.probs(), qml.var(qml.Z(3))]
+        _ = qml.debugging._measure(measurement_process)
+
+    executed_tape = qml.tape.QuantumScript.from_queue(queue)
+    expected_tape = qml.tape.QuantumScript(ops, measurements)
+
+    assert qml.equal(expected_tape, executed_tape)  # no unexpected queuing
+
+    expected_debugging_tape = qml.tape.QuantumScript(ops, measurements + [measurement_process])
+    executed_debugging_tape = mock_method.call_args.args[0][0]
+
+    assert qml.equal(
+        expected_debugging_tape, executed_debugging_tape
+    )  # _execute was called with new measurements
+
+
+@patch.object(PLDB, "_execute")
+def test_state(_mock_method):
+    """Test that the state function works as expected."""
+    with qml.queuing.AnnotatedQueue() as queue:
+        qml.RX(1.23, 0)
+        qml.RY(0.45, 2)
+        qml.sample()
+
+        _ = qml.debugging.state()
+
+    assert qml.state() not in queue
+
+
+@patch.object(PLDB, "_execute")
+def test_expval(_mock_method):
+    """Test that the expval function works as expected."""
+    for op in [qml.X(0), qml.Y(1), qml.Z(2), qml.Hadamard(0)]:
+        with qml.queuing.AnnotatedQueue() as queue:
+            qml.RX(1.23, 0)
+            qml.RY(0.45, 2)
+            qml.sample()
+
+            _ = qml.debugging.expval(op)
+
+        assert op not in queue
+        assert qml.expval(op) not in queue
+
+
+@patch.object(PLDB, "_execute")
+def test_probs_with_op(_mock_method):
+    """Test that the probs function works as expected."""
+
+    for op in [None, qml.X(0), qml.Y(1), qml.Z(2)]:
+        with qml.queuing.AnnotatedQueue() as queue:
+            qml.RX(1.23, 0)
+            qml.RY(0.45, 2)
+            qml.sample()
+
+            _ = qml.debugging.probs(op=op)
```

Comment:
Do we need to do an assignment here? I think it can be skipped.

Reply:
I think I was getting some kind of formatting error, but let's try it.